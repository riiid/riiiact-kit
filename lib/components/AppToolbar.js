import React, {Component, PropTypes} from 'react';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import DescIcon from 'material-ui/svg-icons/action/description';
import ChartIcon from 'material-ui/svg-icons/editor/show-chart';
import ClassIcon from 'material-ui/svg-icons/action/class';
import FeedbackIcon from 'material-ui/svg-icons/action/feedback';

class AppToolbar extends Component {

  constructor(props) {
    super(props);
    this.state = {
    };
  }

  _route(index) {
    const {onOpenSnackbar} = this.props;
    const {router} = this.context;
    switch (index) {
      case 0:
      case 1:
      case 2:
        return router.push('menu');
      case 3:
        return onOpenSnackbar();
      case 100:
        return router.push('/');
      default:
        break;
    }
  }

  _menu() {
    return (
      <IconMenu iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
        targetOrigin={{horizontal: 'right', vertical: 'top'}}
        anchorOrigin={{horizontal: 'right', vertical: 'top'}}>
      <MenuItem primaryText="Menus" disabled={true} />
      <MenuItem primaryText="menu 1"
        leftIcon={<DescIcon/>}
        onTouchTap={this._route.bind(this, 0)}/>
      <MenuItem primaryText="menu 2"
        leftIcon={<ChartIcon/>}
        onTouchTap={this._route.bind(this, 1)}/>
      <MenuItem primaryText="menu 3"
        leftIcon={<ClassIcon/>}
        onTouchTap={this._route.bind(this, 2)}/>
      <MenuItem primaryText="show snackbar"
        leftIcon={<FeedbackIcon/>}
        onTouchTap={this._route.bind(this, 3)}/>
      </IconMenu>
    );
  }

  render() {
    const {className, onToggleDrawer} = this.props;
    return (
      <AppBar style={{position: 'fixed', top: 0, zIndex: 1101}}
        className={className}
        onLeftIconButtonTouchTap={onToggleDrawer}
        onTitleTouchTap={this._route.bind(this, 100)}
        iconElementRight={this._menu()}
        title="Riiiact-Kit"
        titleStyle={{cursor: 'pointer'}}/>
    );
  }
}

AppToolbar.displayName = 'AppToolbar';
AppToolbar.contextTypes = {
  router: PropTypes.object.isRequired
};
AppToolbar.propTypes = {
  className: PropTypes.string,
  onToggleDrawer: PropTypes.func,
  onOpenSnackbar: PropTypes.func
};

export default AppToolbar;
