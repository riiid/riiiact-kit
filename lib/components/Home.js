import React, {Component} from 'react';
import Row from 'react-bootstrap/lib/Row';
import Col from 'react-bootstrap/lib/Col';
import {TITLE, DEPENDENCIES} from 'config';
import map from 'lodash/map';
import {Card, CardTitle, CardText} from 'material-ui/Card';

class Home extends Component {
  render() {
    return (
      <Row>
        <Col xs={10} sm={8} md={8} xsOffset={1} smOffset={2} mdOffset={2}>
          <Card>
            <CardTitle title={TITLE} subtitle="powered by"/>
            <CardText>
              <ul style={{marginTop: 0}}>
              {
                map(DEPENDENCIES, (v, k) => {
                  return (
                    <li key={k}><code>{k}</code> : <code>{v}</code></li>
                  );
                })
              }
              </ul>
            </CardText>
          </Card>
        </Col>
      </Row>
    );
  }
}

Home.displayName = 'Home';

export default Home;
