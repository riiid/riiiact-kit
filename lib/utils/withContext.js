import React, {Component} from 'react';

export default (node, context, childContextTypes) => {
  /* eslint react/display-name: 0 */
  class Wrapper extends Component {
    getChildContext() {
      return context;
    }

    render() {
      return node;
    }
  }
  Wrapper.childContextTypes = childContextTypes;
  return <Wrapper/>;
};
