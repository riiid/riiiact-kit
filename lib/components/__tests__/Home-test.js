import React from 'react';
import renderer from 'react-test-renderer';
import withContext from 'utils/withContext';
import AppToolbar from 'components/AppToolbar';
import Home from 'components/Home';

const {context, childContextTypes} = __OPTIONS;

describe('components/Home', () => {

  it('render', () => {
    const tree = renderer.create(
      withContext(<Home/>, context, childContextTypes)
    );

    expect(tree).toMatchSnapshot();
  });
});

