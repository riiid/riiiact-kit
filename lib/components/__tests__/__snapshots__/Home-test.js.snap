// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Home render 1`] = `
<div
  className="row"
>
  <div
    className="col-md-8 col-md-offset-2 col-sm-8 col-sm-offset-2 col-xs-10 col-xs-offset-1"
  >
    <div
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "zIndex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": 8,
          }
        }
      >
        <div
          style={
            Object {
              "muiPrepared": true,
              "padding": 16,
              "position": "relative",
            }
          }
        >
          <span
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "block",
                "fontSize": 24,
                "lineHeight": "36px",
                "muiPrepared": true,
              }
            }
          />
          <span
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "display": "block",
                "fontSize": 14,
                "muiPrepared": true,
              }
            }
          >
            powered by
          </span>
        </div>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "muiPrepared": true,
              "padding": 16,
            }
          }
        >
          <ul
            style={
              Object {
                "marginTop": 0,
              }
            }
          >
            <li>
              <code>
                package1
              </code>
               : 
              <code>
                VERSION
              </code>
            </li>
            <li>
              <code>
                package2
              </code>
               : 
              <code>
                VERSION
              </code>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
